Sources/Mailgun/Client.swift

```diff
@@ -86,19 +106,18 @@ public struct Client {
 private func setBaseUrl(_ baseUrl: URL) -> (URLRequest) -> URLRequest {
   return { request in
     var request = request
-    request.url = baseUrl.appendingPathComponent(request.url?.relativePath ?? "")
+    request.url = URLComponents(url: request.url!, resolvingAgainstBaseURL: false)?.url(relativeTo: baseUrl)
     return request
   }
 }
 
 private func runMailgun<A>(
   apiKey: Client.ApiKey,
-  domain: Client.Domain,
   logger: Logger
   ) -> (DecodableRequest<A>?) -> EitherIO<Error, A> {
 
   return { mailgunRequest in
-    guard let baseUrl = URL(string: "https://api.mailgun.net/v3/\(domain)")
```

Comment:
apparently sometimes the domain is baked into the API endpoint and sometimes not, so going to leave it to the individual endpoint functions to decide how to construct the url